src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs

```diff
@@ -180,69 +157,76 @@ internal unsafe SocketError DoOperationAccept(Socket socket, SafeCloseSocket han
             NativeOverlapped* overlapped = AllocateNativeOverlapped();
             try
             {
-                int bytesTransferred;
+                bool userBuffer = _count >= _acceptAddressBufferCount;
```

Comment:
Why are we testing this here?  Why aren't we just testing something like _buffer != default?

Reply:
Logically they're equivalent.  StartOperationAccept checks whether _buffer.Equals(default).  If it doesn't, meaning the user supplied a buffer, then it checks to see whether the length is less than the minimum supports for accepts (_count < _acceptAddressBufferCount), throwing if it's too small.  So here, we should be using the user's buffer if it met that criteria, which means either _count == 0 and we should use the accept buffer or _count >= _acceptAddressBufferCount and we should use the user buffer.  We could also check !_buffer.Equal(default), but that's slightly more expensive, and they should be equivalent for this purpose.

Reply:
Ok, then why not test if _count == 0?  This is presumably the cheapest thing to do.

Reply:
Hmm, yeah, that should work, too.